satpy/readers/viirs_vgac_l1c_nc.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2009-2023 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+"""Reading VIIRS VGAC data."""
+
+import logging
+from datetime import datetime
+
+import numpy as np
+import xarray as xr
+
+from satpy import CHUNK_SIZE
+from satpy.readers.file_handlers import BaseFileHandler
+
+logger = logging.getLogger(__name__)
+
+
+class VGACFileHandler(BaseFileHandler):
+    """Reader VGAC data."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init the file handler."""
+        super(VGACFileHandler, self).__init__(
+            filename, filename_info, filetype_info)
+
+        self.engine = "h5netcdf"
+        self._start_time = filename_info['start_time']
+        self._end_time = None
+        self.sensor = 'viirs'
+        self.filename_info = filename_info
+
+    def convert_to_bt(self, data, data_lut, scale_factor):
+        """Convert radances to brightness temperatures."""
+        from scipy import interpolate
+        x = np.arange(0, len(data_lut))
+        y = data_lut
+        func = interpolate.interp1d(x, y)
+        brightness_temperatures = func(data.values / scale_factor)
+        return brightness_temperatures
+
+    def fix_radiances_not_in_percent(self, data):
+        """Scale radiances to percent. This was not done in first version of data."""
+        return 100*data
+
+    def get_dataset(self, key, yaml_info):
+        """Get dataset."""
+        logger.debug("Getting data for: %s", yaml_info['name'])
+        nc = xr.open_dataset(self.filename, engine=self.engine, decode_times=False,
+                             chunks={'y': CHUNK_SIZE, 'x': CHUNK_SIZE})
```

Comment:
For performance purposes, it might be better to use (a multiple of) the native size of the data.
Here is an example of how we do it in another place: https://github.com/pytroll/satpy/blob/main/satpy/readers/avhrr_l1b_gaclac.py#L177-L180, maybe we can do the same here?